lib/src/main/kotlin/RB.kt

```diff
@@ -0,0 +1,526 @@
+package RBTree
+
+import TreeMap.*
+
+enum class Color {
+    RED, BLACK
+}
+
+
+enum class Side {
+    left, right
+}
+
+
+class RBNode<K: Comparable<K>, V>(key: K, value: V, parent: RBNode<K, V>?, color: Color?): Node<K, V, RBNode<K, V>>(key, value, parent){
+    internal var color: Color? = color
+}
+
+
+class RBTree<K: Comparable<K>, V>: TreeMap<K, V, RBNode<K, V>>() {
+    override var root: RBNode<K, V>? = null
+
+    // поворот налево: n - новая "старшая вершина", отец n становится левым ребенком n
+    private fun leftRotation(n: RBNode<K, V>?) {
+        var updateRoot = false
```

Comment:
Можно было обойтись без updateRoot.

```suggestion
         if (n == null) {
            return
        }
        if (n.parent == this.root) {
            this.root = n
        }
```